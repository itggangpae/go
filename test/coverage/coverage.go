package coverage

import "fmt"

// 도달할 수 없는 코드를 가진 함수
func f1() {
	if true {
		fmt.Println("무조건 호출!")
	} else {
		fmt.Println("절대로 호출될 수 없는 코드!") //도달 할 수 없는 코드
	}
}

// 음수를 대입하면 동작하지 않고 양수를 대입하면 첫번째 코드만 수행되는 함수
func f2(n int) int {
	if n >= 0 {
		return 0
	} else if n == 1 {
		return 1
	} else {
		return f2(n-1) + f2(n-2)
	}
}
