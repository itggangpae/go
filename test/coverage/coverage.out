mode: set
/Users/adam/Documents/lecture/Go/source/test/coverage/coverage.go:6.11,7.10 1 1
/Users/adam/Documents/lecture/Go/source/test/coverage/coverage.go:7.10,9.3 1 1
/Users/adam/Documents/lecture/Go/source/test/coverage/coverage.go:9.8,11.3 1 0
/Users/adam/Documents/lecture/Go/source/test/coverage/coverage.go:15.20,16.12 1 1
/Users/adam/Documents/lecture/Go/source/test/coverage/coverage.go:16.12,18.3 1 1
/Users/adam/Documents/lecture/Go/source/test/coverage/coverage.go:18.8,18.19 1 0
/Users/adam/Documents/lecture/Go/source/test/coverage/coverage.go:18.19,20.3 1 0
/Users/adam/Documents/lecture/Go/source/test/coverage/coverage.go:20.8,22.3 1 0
